crates/bevy_ui/src/lib.rs

```diff
@@ -75,6 +78,60 @@ impl Default for UiScale {
     }
 }
 
+/// General representation of progress between two values.
+#[derive(Debug, Clone)]
+pub struct Progress<T>
+where
+    T: Send + Sync + Clone + 'static,
+{
+    /// The minimum value that the progress can have, inclusive.
+    pub min: T,
+    /// The maximum value that the progress can have, inlucsive.
+    pub max: T,
+    /// The current value of progress.
+    pub value: T,
+}
+
+impl<T: Send + Sync + Copy> Progress<T>
+where
+    T: PartialOrd<T>,
+{
+    pub fn new(value: T, min: T, max: T) -> Self {
```

Comment:
Should this function check if `value` is actually in range ?

Reply:
It does, by calling the `from_range()`, which has the check.